src/dotty/tools/dotc/transform/Space.scala

```diff
@@ -166,26 +166,34 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
    *  otherwise approximate extractors to Empty
    */
   def project(pat: Tree, roundUp: Boolean = true)(implicit ctx: Context): Space = pat match {
-    case Ident(_) => Typ(pat.tpe)
-    case Select(_, _) => Typ(pat.tpe)
+    case Ident(_) => Typ(pat.tpe.stripAnnots, false)
+    case Select(_, _) => Typ(pat.tpe.stripAnnots, false)
     case Alternative(trees) => Or(trees.map(pat => project(pat, roundUp)))
     case Bind(_, pat) => project(pat)
     case UnApply(_, _, pats) =>
       if (pat.tpe.classSymbol.is(CaseClass))
-        Kon(pat.tpe, pats.map(pat => project(pat, roundUp)))
-      else if (roundUp) Typ(pat.tpe)
+        Kon(pat.tpe.stripAnnots, pats.map(pat => project(pat, roundUp)))
+      else if (roundUp) Typ(pat.tpe, false)
       else Empty
     case Typed(pat @ UnApply(_, _, _), _) => project(pat)
-    case Typed(expr, _) => Typ(expr.tpe)
+    case Typed(expr, _) => Typ(expr.tpe.stripAnnots, true)
     case _ =>
       debug(s"========unkown tree: $pat========")
       Empty
   }
 
-  def isSubType(tp1: Type, tp2: Type): Boolean = {
-    val res = tp1 <:< tp2
-    debug(s"$tp1 <: $tp2 ? $res")
-    res
+  /** Is `tp1` a subtype of `tp2`?
+   *
+   *  Ignore type parameters in comparison due to erasure, i.e., Some[Int] <: Some[T]
```

Comment:
You might be able to use `TypevarsMissContext` instead (e.g: `tp1.<:<(tp2)(TypevarsMissContext)(ctx.fresh.addMode(Mode.TypevarsMissContext).setExploreTyperState)` instead), but I don't really understand why this is needed in your case